spec/updateUserProfile.ts

```diff
@@ -0,0 +1,206 @@
+      response.should.have.status(HttpStatus.OK);
+    });
+  });
+
+  describe('Update user settings', () => {
+    before(async () => {
+      passportStub = sinon.stub(passport, 'authenticate').returns(async (ctx: any, next: any) => {
+        await next();
+      });
+      await User.create(user);
+    });
+
+    after(async () => {
+      passportStub.restore();
+      await User.remove({ email: user.email });
+    });
+
+    it('should return updated token with new user settings', async () => {
+      const response = await chai
+        .request(app)
+        .post('/api/user/update-user-settings')
+        .set('content-type', 'application/json')
+        .set('authorization', `Bearer ${token}`)
+        .send({
+          interfaceLanguage: 'en',
+          subscription: true
+        });
+      const updatedUser = decodeToken(response.body.token);
+      updatedUser.should.have.property('email').equal(user.email);
+      updatedUser.should.have.property('interfaceLanguage').equal('en');
+      updatedUser.should.have.property('subscription').equal(true);
+      response.should.have.status(HttpStatus.OK);
+    });
+
+    it('should throw error if user does not exist', async () => {
+      const newEmail = 'email@test.com';
+      const userEmail = 'invalid@test.com';
+      try {
+        await sendEmailConfirmation(userEmail, newEmail);
+        assert.fail('No error has been thrown even though such user does not exist');
+      } catch (error) {
+        assert.equal(codeErrors.INCORRECT_EMAIL_OR_PASS, error.data.code);
+      }
+    });
+  });
+
+  describe('Update user image', () => {
+    const imageData = {
+      image: 'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7',
+      type: 'image/gif'
+    };
+
+    before(async () => {
+      passportStub = sinon.stub(passport, 'authenticate').returns(async (ctx: any, next: any) => {
+        await next();
+      });
+      await User.create(user);
+    });
+
+    after(async () => {
+      passportStub.restore();
+      await User.remove({ email: user.email });
+    });
+
+    it('should return updated token with new user image', async () => {
+      const response = await chai
+        .request(app)
+        .post('/api/user/update-user-image')
+        .set('content-type', 'application/json')
+        .set('authorization', `Bearer ${token}`)
+        .send(imageData);
+      const updatedUser = decodeToken(response.body.token);
+      updatedUser.should.have.property('email').equal(user.email);
+      updatedUser.should.have.property('image').equal(updatedUser.image);
+      response.should.have.status(HttpStatus.OK);
+    });
+
+    it('should return an object with valid url', async () => {
+      const email = 'email@test.com';
+      const res = await updateImage(email, imageData);
+      expect(res).to.have.all.keys('image', 'type');
+      expect(res.image).to.include('https://carfinder-bucket.s3.us-east-2.amazonaws.com');
```

Comment:
Please don't hardcode such variables. This should go from the ENV.
Or use some kind of a regex. Say the string should contain s3 and amazonaws. Something like that.

Reply:
Fixed: replaced with regex